### ISSUE TITLE
HTTP Error 404.15 when running a lengthy query

### ISSUE THREAD

**Issue by MikeAlhayek** [OrchardCMS/OrchardCore#12840]
### Describe the bug
When created a SQL Query that is very lengthy and you click `Run`, you get 404 error because the generated URL is too long. 

### To Reproduce
1. Enable `SQL Queries` feature
2. Go to Search >> Queries >> All queries
3. Click on `Add Query`
4. Provides a long query in the `Query` field
5. Save
6. Now you should be on the list of queries view, click `Run` next to the lengthy query you created. You should get the error, if not increase the length of the query and try again.

### Expected behavior
I expect the query to run with no issues.

Instead of passing the actual query as Base64 string to the query, maybe we should send the name of the query. Then when the query runner view open, can can get the query by name and load the page correctly.

**Comment by Skrypt** [comment]
You can also set the max length of your form posts on your server so that it passes. Also, there is a limitation of length to a Query passed to SQL Server.

**Comment by MikeAlhayek** [comment]
Yes. But you should not have to increase the limits of your server just to test out your query. We can easily remove this limitation by passing the name of the query instead of the encoded query itself. 

Yes SQL server has its own limitation, that isn't something we can control. But from UI perspective, we should about this error

**Comment by Skrypt** [comment]
It is an API call so you pass the Query and you get the result. So, we would need a different API call for getting results from a Named Query. Though, named Queries are used for Lucene and SQL and ElasticSearch...

**Comment by sebastienros** [comment]
We could use a POST then and pass the query in the body.
